-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-34962/24792

ERROR: Build errors found during clang-tidy run.
```
RecoHGCal/TICL/plugins/TrackstersProducer.cc:58:48: error: cannot initialize a member subobject of type 'ticl::Trackster::IterationIndex' with an rvalue of type 'int' [clang-diagnostic-error]
  ticl::Trackster::IterationIndex iterIndex_ = 0;
                                               ^
Suppressed 2389 warnings (2385 in non-user code, 3 NOLINT, 1 with check filters).
--
gmake: *** [config/SCRAM/GMake/Makefile.coderules:129: code-checks] Error 2
gmake: *** [There are compilation/build errors. Please see the detail log above.] Error 2
```

--- a/RecoHGCal/TICL/plugins/TrackstersProducer.cc
+++ b/RecoHGCal/TICL/plugins/TrackstersProducer.cc
@@ -55,7 +55,7 @@ private:
   edm::EDGetTokenT<TICLLayerTilesHFNose> layer_clusters_tiles_hfnose_token_;
   const edm::EDGetTokenT<std::vector<TICLSeedingRegion>> seeding_regions_token_;
   const std::string itername_;
-  ticl::Trackster::IterationIndex iterIndex_ = 0;
+  ticl::Trackster::IterationIndex iterIndex_ = ticl::Trackster::IterationIndex(0);
 };
 DEFINE_FWK_MODULE(TrackstersProducer);
 

